### PR TITLE
Tgp 1924/create a new endpoint in data store to clear the db collections

### PR DESCRIPTION
**Pull request by pauloncast**

### CODE REVIEW

File: test/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileControllerSpec.scala [hunk 1]
```diff
@@ -186,4 +188,23 @@ class ProfileControllerSpec extends SpecBase with MockitoSugar {
     }
 
   }
+
+  s"PUT $deleteAllUrl" - {
+
```
Comment (yohan-oc) [inline test/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileControllerSpec.scala:193]: DELETE ?

**Review by AmarkumarJagtap — APPROVED** [review]

**Review by liandra-giacomine — CHANGES_REQUESTED** [review]
We shouldn't be adding functionality in here to facilitate tests, this can be done in the performance test itself

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileController.scala [hunk 2]
```diff
@@ -62,4 +64,11 @@ class ProfileController @Inject() (
       }
   }
 
+  def deleteAll(): Action[AnyContent] = identify.async {
+    if (config.droppingProfileCollection)
+      profileRepository.deleteAll.map(_ => Ok)
+    else
+      Future.successful(NotFound)
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileController.scala:71]: not sure about returning not found here, maybe internals server error?

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/ProfileRepository.scala [hunk 3]
```diff
@@ -61,4 +61,7 @@ class ProfileRepository @Inject() (
       .map(_ => true)
   }
 
+  def deleteAll: Future[Boolean] =
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/ProfileRepository.scala:64]: can you return Future[Done] here, a boolean doesn't represent anything here

**Review by liandra-giacomine — APPROVED** [review]

**Pull request merged**